bin/node/runtime/src/lib.rs

```diff
@@ -1240,6 +1241,7 @@ impl pallet_contracts::Config for Runtime {
 	type MaxStorageKeyLen = ConstU32<128>;
 	type UnsafeUnstableInterface = ConstBool<false>;
 	type MaxDebugBufferLen = ConstU32<{ 2 * 1024 * 1024 }>;
+	type Migrations = (NoopMigration<1>, NoopMigration<2>);
```

Comment:
A `()` should suffice.

Reply:
these NoopMigrations are used by benchmarks, so we should keep them here

https://github.com/paritytech/substrate/blob/28f56b6938f8a5be7bc7d6b94cba5a919b59545d/frame/contracts/src/benchmarking/mod.rs#L281-L290

Reply:
Then how does this look like in Polkadot?  
Could be as ugly as [this](https://github.com/paritytech/polkadot/blob/81309d1469bea4f98d209c6c9ba5d57e6ff0bb60/runtime/polkadot/src/lib.rs#L1161)...

Reply:
> I think the things used for tests & benches should not appear here. It's better to keep them where they were, behind a feature flag. That's being said the docs giving an example how to configure the Migrations seq in runtime Config, would be great to have.

@sacha-l  I think that's part of the config, I think we can't avoid having it here. Compile flags worked before because the type was internal to the pallet.

Reply:
> Then how does this look like in an actual runtime? Could be as ugly as [this](https://github.com/paritytech/polkadot/blob/81309d1469bea4f98d209c6c9ba5d57e6ff0bb60/runtime/polkadot/src/lib.rs#L1161)...

We could do something as ugly as that or as ugly as this (which was the initial approach)
```rust
/// Performs all necessary migrations based on `StorageVersion`.
#[cfg(not(any(feature = "runtime-benchmarks")))]
pub struct Migration<T: Config, M: MigrateSequence = T::Migrations>(PhantomData<(T, M)>);

/// Custom migration for running runtime-benchmarks and tests.
#[cfg(any(feature = "runtime-benchmarks"))]
pub struct Migration<T: Config, M: MigrateSequence = (NoopMigration<1>, NoopMigration<2>)>(
	PhantomData<(T, M)>,
);
```

Reply:
I like how it is setup in @ggwpez 's link at least it makes it clear that these noopMigration are intended for benchmarks

Reply:
I have applied PG's suggested fix https://github.com/paritytech/substrate/pull/14309#discussion_r1221429649